src/systems/filecoin_markets/storage_market/storage_market_actor.id

```diff
@@ -17,119 +16,96 @@ type StorageMarketActorState struct {
+        fromAddr         addr.Address
+        amountRequested  actor.TokenAmount
+    ) actor.BalanceTableHAMT
 
-    _terminateDeal(rt Runtime, dealID deal.DealID) actor.TokenAmount
-    _rtSlashDealCollateral(
-        rt      Runtime
-        amount  actor.TokenAmount
-    ) actor.TokenAmount
+    _rtTransferBalance(
+        rt                       Runtime
+        fromAddr                 addr.Address
+        toAddr                   addr.Address
+        transferAmountRequested  actor.TokenAmount
+    )
 
-    _refundStartedDeals(rt Runtime, dealIDs [deal.DealID])
-    _addressEntryExists(address addr.Address) bool
+    _rtSlashBalance(rt Runtime, addr addr.Address, slashAmount actor.TokenAmount)
 }
 
 type StorageMarketActorCode struct {
-    WithdrawBalance(rt Runtime, balance actor.TokenAmount)
-    AddBalance(rt Runtime)  // amount is in the message
-
-    // call by StorageMiningSubsystem before PreCommitSector
-    // a StorageDeal is only published on chain when it passes _validateNewStorageDeal
-    // _validateNewStorageDeal checks for the following:
-    //     - deal did not start/expire when it is signed
-    //     - deal hits the chain before its StartEpoch
-    //     - client and provider address and signature are correct (TODO may be unnecessary)
-    //     - client and provider have sufficient balances to lock up
-    // a DealID will be assigned and stored in the mapping of DealID to StorageDeal
-    // _lockBalance is then called after each _validateNewStorageDeal to lock up balances
-    PublishStorageDeals(rt Runtime, deals [deal.StorageDeal]) [PublishStorageDealResponse]
-
-    // Call by sm.PreCommitSector
-    // Verify the following:
-    //     - dealIDs have been registered in PublishedDeals
-    //     - deals will not expire by the time ProveCommit hits
-    //     - deals have not started
-    // Fail PreCommit if any of above is not true and abort
-    VerifyPublishedDealIDs(rt Runtime, dealIDs [deal.DealID])
-
-    // Call by sm.ProveCommitSector
-    // Verify the following:
-    //     - dealIDs are in PublishedDeals
-    //     - deals have not expired, OK if deals have started
-    // Fail ProveCommit if any of the above is not true and abort
-    // Move deals from PublishDeals to ActiveDeals if all deals satisfy the requirement
-    ActivateDeals(rt Runtime, dealIDs [deal.DealID]) [deal.OnChainDeal]
-
-    // call by StorageMinerActor
-    // slash deal collateral depending on the faults (only TerminatedFault)
-    ProcessDealSlash(
-        rt         Runtime
-        dealIDs    [deal.DealID]
-        faultType  sector.StorageFaultType
+    // Deposits the specified amount into the balance held in escrow.
+    // Note: the amount is included implicitly in the message.
+    AddBalance(rt Runtime)
+
+    // Attempt to withdraw the specified amount from the balance held in escrow.
+    // If less than the specified amount is available, yields the entire available balance.
```

Comment:
No -- it should return nil (serialized in `InvokeMethod`) to indicate success. We might consider returning the amount actually withdrawn, although then this would use up space on-chain, and it can be inferred from the previous chain state anyway.